js/views/LoginPage.js

```diff
@@ -10,60 +11,127 @@ import React, {
+    this.props.setTitle(this.title);
+  },
+
   getInitialState() {
     return {
-      emailInput: '',
-      passwordInput: '',
-      button_text: 'Verify',
+      username: '',
+      password: '',
+      button_text: 'Login',
+      errors: [],
     }
   },
 
   /* Methods */
-  handleVerifyLogin(event) {
-    event.preventDefault()
+  handleVerifyLogin() {
+    let self = this;
+
+    // validate
+    let errors = this.joiValidate();
+    if (errors.length > 0) {
+      Alert.alert('Validation', 'The form errors need corrected to continue.');
+      return;
+    }
+
+    let state = Object.assign({}, this.state);
+    state.button_text = 'Verifying...';
+    this.setState(state);
 
-    this.setState({
-      button_text: 'Verifying...'
-    })
+    authenticate(state.username, state.password, function(err, user) {
+      if (err) {
+        // reset button_text state
+        state.button_text = 'Login';
+        self.setState(state);
+        // show a message
+        Alert.alert(err, 'The username and password combination is invalid.')
+        return;
+      }
+      // the user is authenticated, setAuthenticated to true
+      self.props.setAuthenticated(true)
+      // allow for the async state to be updated
+      async.nextTick(function(){
+        // navigate to the default route
+        self.props.navigator.replace({});
+      });
+    });
+  },
 
-    verifyLogin(this.state.text, this.handleVerifyLoginResponse)
+  decodeText(txt) {
+    if (txt) {
+      return he.decode(txt);
+    }
+    return '';
   },
 
-  handleVerifyLoginResponse(response) {
-    this.setState({
-      button_text: 'Login Verified'
-    })
+  textFieldChangeHandler(name, text) {
+    let schema = {};
+    schema[name] = this.schema[name];
+    let object = {};
+    object[name] = text;
+    this.joiCheckError(object, schema);
+    let state = Object.assign({}, this.state);
+    state[name] = text;
+    this.setState(state);
   },
 
   /* Render */
   render() {
     return (
       <View style={Styles.container.default}>
-        <TextInput
-          style={Styles.form.input}
-          onChangeText={(emailInput) => this.setState({emailInput})}
-          value={this.state.emailInput}
-          placeholder="Login"
-        />
-        <TextInput
-          style={Styles.form.input}
-          onChangeText={(passwordInput) => this.setState({passwordInput})}
-          value={this.state.passwordInput}
-          placeholder="Password"
-        />
-        <TouchableHighlight onPress={this.handleVerifyLogin}>
-          <Text>
-            {this.state.button_text}
+        <View style={Styles.form.inputGroup}>
+          <Text style={Styles.form.errorText}>
+            {this.decodeText(this.state.errors.username)}
+          </Text>
+          <TextInput
+            style={Styles.form.input}
+            onChangeText={this.textFieldChangeHandler.bind(this, 'username')}
+            value={this.state.username}
+            placeholder="Username"
```

Comment:
Minor nuisance affecting iOS devices: It will try to auto-capitalize and auto-correct words by default. Login is case-sensitive and these may cause issues logging in.
These can be disabled with the props:

autoCapitalize='none'
autoCorrect={false}